core/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -104,7 +105,7 @@ protected void masterOperation(final GetSnapshotsRequest request, ClusterState s
                     }
                 }
                 for (String snapshot : snapshotsToGet) {
-                    SnapshotId snapshotId = new SnapshotId(request.repository(), snapshot);
+                    final SnapshotId snapshotId = snapshotsService.resolveSnapshotId(new SnapshotName(request.repository(), snapshot));
```

Comment:
Although, to really make it worth it efficiency wise, the Repository interface should implement it as well, so we are only reading from disk once.